api/pom.xml

```diff
@@ -50,7 +50,10 @@
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>kafka-clients</artifactId>
-            <version>${kafka-clients.version}</version>
```

Comment:
@Haarolean 

The significant version difference is because Confluent follows its own versioning strategy, but ultimately uses the open-source distribution binaries as a foundation. For example, Confluent Platform `v7.8.0` is built on Kafka `v3.8.x`. See this page for more details: https://docs.confluent.io/platform/current/installation/versions-interoperability.html.

In general, it's best to choose one version or the other, as using both together can lead to binary incompatibility issues. For instance, in the current main branch, we're using client versions that pull v3.8.x while also pulling Kafka 3.4.

The only valid reason for using a separate version for the clients package would be if you need to upgrade it independently while delaying the upgrade of other packages. However, this is a limited use case and not really worth the complexity.

For most users, it's better to stick to one version. Mixing them offers few benefits and only increases the workload, as each version will need to be upgraded separately, and you need to be cautious not to mix them up

There are some nices to have that the confluent distribution brings us, but we should

Reply:
Thanks for the explanation. My concern was about introducing incompatible changes with stuff like our serde API, hope that's not the case here.